src/database.c

```diff
@@ -453,6 +453,13 @@ hs_error_t print_database_string(char **s, u32 version, const platform_t plat,
 HS_PUBLIC_API
 hs_error_t hs_serialized_database_info(const char *bytes, size_t length,
                                        char **info) {
+    // Decode and check the header
```

Comment:
This check should happen after we know we have enough bytes to decode the header, i.e. after [line 470](https://github.com/starius/hyperscan/blob/7561276ca79e6fa406e2ed01445bac0a9a487c71/src/database.c#L470).